store/store.go

```diff
@@ -683,13 +683,13 @@ func (s *Store) Apply(l *raft.Log) interface{} {
 	}
 }
 
-// Snapshot returns a snapshot of the database. The caller must ensure that
+// RawDBSnapshot returns a snapshot of the database. The caller must ensure that
 // no transaction is taking place during this call. Hashicorp Raft guarantees
```

Comment:
The docstring for this function shouldn't contain any reference to Hashicorp Raft, it's not relevant in this context. This is function that is handing control over to the user, and they better be sure they know what they are doing.

There are multiple potential ways to shoot yourself in the foot here. There is no check that the store is the leader of the cluster, for example, and therefore no guarantee that the copy you get is up-to-date.